tasks/change-granule-collection-s3/tests/test-index.js

```diff
@@ -0,0 +1,1000 @@
+}
+
+function buildPayload(t, collection) {
+  const newPayload = t.context.payload;
+  newPayload.config.targetCollection = collection;
+  newPayload.config.bucket = t.context.stagingBucket;
+  newPayload.config.buckets.internal.name = t.context.stagingBucket;
+  newPayload.config.buckets.public.name = t.context.publicBucket;
+  newPayload.config.buckets.private.name = t.context.privateBucket;
+  newPayload.config.buckets.protected.name = t.context.protectedBucket;
+  return newPayload;
+}
+
+test.beforeEach(async (t) => {
+  const topicName = randomString();
+  const { TopicArn } = await createSnsTopic(topicName);
+  process.env.granule_sns_topic_arn = TopicArn;
+  const testDbName = `change-granule-collection-s3/change-collections-s3${cryptoRandomString({ length: 10 })}`;
+  changeGranuleCollectionS3 = proxyquire(
+    '../dist/src',
+    {
+      '@cumulus/api-client/granules': {
+        getGranule: (params) => dummyGetGranule(params.granuleId, t),
+      },
+      '@cumulus/api-client/collections': {
+        getCollection: (params) => (
+          dummyGetCollection(params.collectionName, params.collectionVersion)
+        ),
+      },
+    }
+  ).changeGranuleCollectionS3;
+
+  t.context.publicBucket = randomId('public');
+  t.context.protectedBucket = randomId('protected');
+  t.context.privateBucket = randomId('private');
+  t.context.systemBucket = randomId('system');
+  t.context.stackName = 'changeGranuleCollectionS3TestStack';
+  const bucketMapping = {
+    public: t.context.publicBucket,
+    protected: t.context.protectedBucket,
+    private: t.context.privateBucket,
+    system: t.context.systemBucket,
+  };
+  t.context.bucketMapping = bucketMapping;
+  await Promise.all([
+    s3().createBucket({ Bucket: t.context.publicBucket }),
+    s3().createBucket({ Bucket: t.context.protectedBucket }),
+    s3().createBucket({ Bucket: t.context.privateBucket }),
+    s3().createBucket({ Bucket: t.context.systemBucket }),
+  ]);
+  process.env = {
+    ...process.env,
+    PG_DATABASE: testDbName,
+    DISTRIBUTION_ENDPOINT: 'https://something.api.us-east-1.amazonaws.com/',
+    system_bucket: t.context.systemBucket,
+    stackName: t.context.stackName,
+  };
+  await putJsonS3Object(
+    t.context.systemBucket,
+    getDistributionBucketMapKey(t.context.stackName),
+    {
+      [t.context.publicBucket]: t.context.publicBucket,
+      [t.context.privateBucket]: t.context.privateBucket,
+      [t.context.protectedBucket]: t.context.protectedBucket,
+      [t.context.systemBucket]: t.context.systemBucket,
+    }
+  );
+});
+
+test.afterEach.always(async (t) => {
+  await recursivelyDeleteS3Bucket(t.context.privateBucket);
+  await recursivelyDeleteS3Bucket(t.context.publicBucket);
+  await recursivelyDeleteS3Bucket(t.context.protectedBucket);
+  await recursivelyDeleteS3Bucket(t.context.systemBucket);
+});
+
+test.serial('Should move files to final location with cmr xml file', async (t) => {
```

Comment:
Nit: test titles should probably include thing being tested (e.g. changeGranuleCollectionS3 should....)

Reply:
Also 'copy files'